### PR TITLE
test: migrate install* and inheritable_workspace_fields to snapbox

### PR DESCRIPTION
**Pull request by eth3lbert**
### What does this PR try to resolve?

Part of https://github.com/rust-lang/cargo/issues/14039.

Migrate following to snapbox:

- `tests/testsuite/inheritable_workspace_fields.rs`
- `tests/testsuite/install.rs`
- `tests/testsuite/install_upgrade.rs`

### CODE REVIEW

File: tests/testsuite/install.rs [hunk 1]
```diff
@@ -2064,7 +2088,22 @@ fn git_install_reads_workspace_manifest() {
...
+  |                           ^
+  |
+[ERROR] invalid type: integer `3`, expected a boolean
+ --> home/.cargo/git/checkouts/foo-[HASH]/[..]/Cargo.toml:6:27
+  |
+6 |             incremental = 3
+  |                           ^
+  |
+
+"#]])
```
Comment (eth3lbert) [inline tests/testsuite/install.rs:2106]: I'm not sure if the duplicated error message shown here is correct. Therefore, I'm currently avoiding redacting with glob intentionally.

File: tests/testsuite/install.rs [hunk 2]
```diff
@@ -1997,8 +2001,11 @@ fn install_global_cargo_config() {
 
     cargo_process("install bar")
         .with_status(101)
-        .with_stderr_contains("[INSTALLING] bar v0.0.1")
-        .with_stderr_contains("[..]--target nonexistent[..]")
+        .with_stderr_data(str![[r#"
+...
```
Comment (weihanglo) [inline tests/testsuite/install.rs:2005]: `[INSTALLING] bar v0.0.1` is missing, though I think that is fine.

File: tests/testsuite/install.rs (hunk 1 above)
Comment (weihanglo) [reply tests/testsuite/install.rs:2106]: CI passes so yeah keep it.

File: tests/testsuite/install.rs [hunk 3]
```diff
@@ -114,74 +113,24 @@ fn simple_with_message_format() {
     pkg("foo", "0.0.1");
 
     cargo_process("install foo --message-format=json")
-        .with_stderr(
-            "\
-[UPDATING] `[..]` index
+        .with_stderr_data(str![[r#"
```
[the remark this answers is not in the document]
Comment (eth3lbert) [reply tests/testsuite/install.rs:116]: Absolutely!  That makes things a lot easier, to be honest. 😅

File: tests/testsuite/install.rs (hunk 2 above)
Comment (eth3lbert) [reply tests/testsuite/install.rs:2005]: I'll double-check this and re-add it.

**Comment by eth3lbert** [comment]
Feedback addressed! Thanks!

**Review by weihanglo — APPROVED** [review]

**Comment by weihanglo** [comment]
Thanks!

@bors r+

**Pull request merged**